groupsforsections.py

```diff
@@ -255,7 +283,25 @@ def main():
 
         # mapping all the users in a sections to corresponding group
         groups_to_users_dict[group_id] = users
-        # TODO: Adding users to corresponding groups
+
+    failed_groups_to_users_dict={}
+    
+    # adding users to the group
+    for group in groups_to_users_dict:
+        for user in groups_to_users_dict[group]:
+            membership_id = add_users_to_group(groups_for_section_class, group, user)
+            if membership_id is None:
+                logging.error('The user %s is not added to the group %s' % (user, group))
+                failed_groups_to_users_dict.setdefault(group, []).append(user)
+            else:
+                logging.info('The User %s got added to the Group %s  with membership id  %s ' %(user, group, str(membership_id)))
+
+    # log the users list that was not added to a group
+    if failed_groups_to_users_dict :
+        logging.error("Some of the users are not added to the groups")
+        for group in failed_groups_to_users_dict:
+            users = ','.join(failed_groups_to_users_dict[group])
+            logging.info('%d users are not added in the group %s and they are %s ' % (len(failed_groups_to_users_dict[group]),group, users))
 
     logging.info('script ran successfully')
```

Comment:
Would it be useful to add a summary count of group / member adds (and failures)?